.github/workflows/github-release.yml

```diff
@@ -0,0 +1,60 @@
+##
+## Create the github release and store artifact files (with checksum)
+##
+
+name: Github Release
+
+# Only run when:
+#   - manually triggered via the ci.yml workflow
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        required: true
+        type: string
+      arch:
+        description: "Stringified JSON object listing of platform matrix"
+        required: true
+        type: string
+    secrets:
+      GH_TOKEN:
+        required: true
+
+jobs:
+  create-release:
+    if: ${{ inputs.tag != '' }}
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Artifacts
+        id: download_artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: artifact
+          path: release
+      - name: Generate Checksums
+        id: generate_checksum
+        uses: jmgilman/actions-generate-checksum@v1
+        with:
+          output: CHECKSUMS.txt
+          patterns: |
+            release/*.zip
+      - name: Display artifact files
+        id: list_artifacts
+        run: ls -R
+        working-directory: release
+      - name: Upload Release
+        id: upload_release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        with:
+          name: Release ${{ github.event.inputs.tag || github.ref }}
+          tag_name: ${{ github.event.inputs.tag || github.ref }}
+          draft: false
+          prerelease: true
```

Comment:
Is this intentional? Should this be parameterized?

Reply:
It definitely could - what would you suggest here, something like `prerelase: false` if a tag is supplied when the workflow is triggered?
for this step, it's really just a copy/paste from the current workflow in a new file: https://github.com/stacks-network/stacks-blockchain/blob/6c5349b0422c5c88db02d578720467c16b102ab2/.github/workflows/ci.yml#L257-L269

The action was updated since the current one in the existing workflow has been deprecated, and the new one has slightly different options. otherwise i kept this logic as unchanged as possible from the current workflow. 

https://github.com/actions/create-release
https://github.com/softprops/action-gh-release/blob/master/action.yml

Reply:
@CharlieC3 ping! did you have an idea that would make sense to change here, or perhaps an idea for an issue for a future iteration?

Reply:
I think `prerelease` should be set to `false` by default, unless this was an intended outcome and part of the release process involves manually changing a prerelease to a release afterwards. This part should be considered for this PR I think.

Ultimately if prereleases are going to be used, I think it would be handy to add a new boolean input to this workflow called `prerelease`, and the value of the boolean would get placed here. This part doesn't have to go in this PR though.

Reply:
personally, i agree here - but it would also be a change in the current process that i wasn't planning to add in this PR: https://github.com/stacks-network/stacks-blockchain/blob/develop/.github/workflows/ci.yml#L283

@kantai @jcnelson any opinion here? the end result to changing this bool would be that when a tagged release is built, a release with binaries that is not set to "pre-release" will also be published. 

I'm in agreement with @CharlieC3 that we should change this from `prerelease: true` -> `prerelease: false` for when a github release is created (only happens on tagged workflows).

Reply:
> The same workflow is used for tagging release candidates (which are pre-releases) and creating feature builds (also pre-releases).

Got it. In that case defaulting on `prerelease: true` makes much more sense to me as well. Should a new input be added in the future, I would think it should continue to create a prerelease by default unless acted upon.

Reply:
I'll add to the issue for the next iteration of work here - there may be a way to get this working without having an extra input or having to manually change that pre-release label.